### PR TITLE
Review

### PR DESCRIPTION
**Pull request by aherlihy**
Hi,

Did you consider using a vue react package? I don't remember if you mentioned that or not. Have a demo on the 'react-vis' branch but it might not be worth it.

I tried to keep style changes out of the first commit, then for the second commit I ran an auto-linter so it should've fixed all the eslint errors. I played around with it and it looked like it had the same behavior, but worth double checking the changes to make sure it didn't do anything crazy.

The only other thing I'm trying to fix is the "check" script. Getting some installation errors, not sure why.

### CODE REVIEW

File: electron/renderer/index.js [hunk 1]
```diff
@@ -49,22 +49,24 @@ render(CompassProfilerVisualizePlugin);
 import Connection from 'mongodb-connection-model';
 import DataService from 'mongodb-data-service';
 
+const DB = 'db1';
+
 const connection = new Connection({
   hostname: '127.0.0.1',
-  port: 27000,
```
Comment (TomerYakir) [inline electron/renderer/index.js]: My local instance is on 27000. Could we keep it as is?

File: src/components/profiler-status/profiler-status.jsx [hunk 2]
```diff
@@ -1,85 +1,63 @@
 import React, { Component } from 'react';
-import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import styles from './profiler-status.less';
 import { Radio, Modal, Button } from 'react-bootstrap';
-import '../../../node_modules/react-bootstrap-modal/lib/css/rbm-patch.css';
+import 'react-bootstrap-modal/lib/css/rbm-patch.css';
```
Comment (TomerYakir) [inline src/components/profiler-status/profiler-status.jsx:6]: I think this can be removed

File: package.json [hunk 3]
```diff
@@ -119,6 +119,7 @@
   "dependencies": {
     "percentile": "^1.2.0",
     "react-bootstrap-table": "^4.3.1",
+    "react-bootstrap-modal":"^4.0.0",
```
Comment (TomerYakir) [inline package.json:122]: I'm going to remove that

File: src/components/profiler-status/profiler-status.jsx [hunk 4]
```diff
@@ -1,85 +1,63 @@
...
-    this.noop();
-  }
-
-  // A no-operation so that the linter passes for the compass-plugin template,
-  // without the need to an ignore rule, because we want the linter to fail when this
-  // dependency is "for-real" not being used (ie: in an actual plugin).
-  noop = () => {
-    const node = ReactDOM.findDOMNode(this);
-    return node;
+    operationThreshold: 15
```
Comment (TomerYakir) [inline src/components/profiler-status/profiler-status.jsx:25]: should be 100

File: src/components/profiler-status/profiler-status.jsx [hunk 5]
```diff
@@ -1,85 +1,63 @@
...
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    if (this.props != prevProps) {
-      this.setState({
-        toProfilerLevel: this.props.profilerLevel,
-        toThreshold: this.props.operationThreshold
-      })
-    }
+  componentWillReceiveProps(nextProps) {
```
Comment (TomerYakir) [inline src/components/profiler-status/profiler-status.jsx:37]: Thanks for that - it felt I was doing it wrong :)

File: src/stores/store.js [hunk 6]
```diff
@@ -19,174 +17,168 @@ const CompassProfilerVisualizeStore = Reflux.createStore({
...
+   *
+   * @return {Object} initial store state.
+   */
+  getInitialState() {
+    return {
+      topQueries: [],
+      slowQueriesOverTime: [],
+      selectedQueries: [],
+      profilerLevel: -1,
+      operationThreshold: 15,
```
Comment (TomerYakir) [inline src/stores/store.js:50]: should be 100

File: electron/renderer/index.js [hunk 7]
```diff
@@ -49,22 +49,24 @@ render(CompassProfilerVisualizePlugin);
...
-  ns: 'euphonia',
```
Comment (aherlihy) [inline electron/renderer/index.js]: sorry you can change this back, I just set it for my own testing

File: src/components/compass-profiler-visualize/compass-profiler-visualize.jsx [hunk 8]
```diff
@@ -10,49 +10,61 @@ class CompassProfilerVisualize extends Component {
...
-    topQueries: []
+    topQueries: [],
+    operationThreshold: 15,
+    profilerLevel: -1,
+    slowQueriesOverTime: [],
+    selectedQueries: [],
+    error: null
   };
 
-  setCurrentQuery = (selectedQueries) => {
```
Comment (aherlihy) [inline src/components/compass-profiler-visualize/compass-profiler-visualize.jsx:21]: anywhere you can use a function you can pretty much use actions

File: electron/renderer/index.js (hunk 1 above)
Comment (aherlihy) [reply electron/renderer/index.js]: Sure, just document it so that people running your plugin won't be confused.

File: package.json (hunk 3 above)
Comment (aherlihy) [reply package.json:122]: your going to remove the modal entirely? Because I couldn't install the plugin without adding that.

**Pull request merged**